### PR TITLE
[React incubation] fixing browse routing

### PR DESCRIPTION
**Pull request by gabe-lyons**
This regression was introduced by https://github.com/linkedin/datahub/pull/2060

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by cptran777** [comment]
I'm confused as to how this wasn't caught in the previous PR. Is the application not throwing the typescript error when you try to build it?

**Comment by gabe-lyons** [comment]
@cptran777 you're correct- it does throw a typescript error on build. Not sure exactly how it snuck in, it rode along in the last commit https://github.com/linkedin/datahub/pull/2060/commits/c79fb8b724fb8e96c6a4092991313faba3c958ff. We're going to set up some basic CI soon so issues like this won't be able to sneak by due to human error in the future.

**Comment by jplaisted** [comment]
There's a build-and-test action that runs on every commit of every PR and just runs `./gradlew build`. Which, afaik, builds everything, right? 

https://github.com/linkedin/datahub/blob/master/.github/workflows/build-and-test.yml

**Comment by gabe-lyons** [comment]
@jplaisted thanks for directing me to that. In that case we have two options

1) extend the build task to build the react app as well. Upsides: we dont need to add a new action. Downsides: if the react app CI breaks, it affects non-react prs

2) create a new action parallel to build-and-test that only runs on changes to datahub-frontend-react. Upsides: breakages in react won't affect non-react prs. Downsides: we have two CI actions.

Thoughts?

**Comment by shirshanka** [comment]
My 2c : let's not allow the react app to be broken. 
So include the react app in the main gradle build.

**Comment by jplaisted** [comment]
To be clear with gradle just add it to the `settings.gradle` file. It'll include it as a submodule, knowing it should build it and also treat it as a module in intellij, etc

**Comment by cptran777** [comment]
I agree, that even in incubation if there's such an obvious issue as "the react app doesn't build properly" that should prevent a PR from being merged.

**Comment by gabe-lyons** [comment]
@jplaisted great- will do that in a followup.

**Comment by shirshanka** [comment]
@gabe-lyons Maybe add it to this PR, so we know this PR is safe to merge ;)

**Comment by gabe-lyons** [comment]
@shirshanka good idea- added

File: settings.gradle.bak [hunk 1]
```diff
@@ -0,0 +1,29 @@
+include 'datahub-dao'
```
Comment (cptran777) [inline settings.gradle.bak:1]: Is this file intentional?

**Review by cptran777 — CHANGES_REQUESTED** [review]

**Comment by gabe-lyons** [comment]
@cptran777 removed 👍

**Review by cptran777 — APPROVED** [review]

**Review by shirshanka — APPROVED** [review]
LGTM! Thanks for reviewing @cptran777 :)

**Pull request merged**